dhis-2/dhis-services/dhis-service-dxf2/src/test/java/org/hisp/dhis/dxf2/events/trackedentity/store/DefaultAclStoreTest.java

```diff
@@ -0,0 +1,181 @@
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.hisp.dhis.IntegrationTestBase;
+import org.hisp.dhis.common.IdentifiableObjectManager;
+import org.hisp.dhis.program.Program;
+import org.hisp.dhis.user.User;
+import org.hisp.dhis.user.UserGroup;
+import org.hisp.dhis.user.UserService;
+import org.hisp.dhis.user.sharing.UserAccess;
+import org.hisp.dhis.user.sharing.UserGroupAccess;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class DefaultAclStoreTest
+    extends IntegrationTestBase
+{
+
+    @Autowired
+    private UserService _userService;
+
+    @Autowired
+    private DefaultAclStore aclStore;
+
+    @Autowired
+    IdentifiableObjectManager manager;
+
+    private User owner;
+
+    private User user;
+
+    @Before
+    public void setUp()
+    {
+        // DhisConvenienceTests needs it to be injected/set for createUser
+        userService = _userService;
+        user = createUser( "userWithoutUserGroup" );
+        owner = createUser( "owner" );
+    }
+
+    @Test
+    public void getAccessibleProgramsReturnsNoneIfNoneIsPublicAndUserHasNoAccess()
+    {
+
+        // a private program
+        Program programA = createProgram( 'A' );
+        programA.setPublicAccess( "--------" );
+        programA.getSharing().setOwner( owner );
+        manager.save( programA, false );
+        // a private program readable by a user group of which the user is NOT
+        // part of
+        Program programB = createProgram( 'B' );
+        programB.setPublicAccess( "--------" );
+        programB.getSharing().setOwner( owner );
+        Set<User> users = new HashSet<>();
+        users.add( owner );
+        UserGroup g = createUserGroup( 'B', users );
+        UserGroupAccess a = new UserGroupAccess();
+        a.setUserGroup( g );
+        a.setAccess( "--r-----" );
+        programB.getSharing().addUserGroupAccess( a );
+        manager.save( programB, false );
+
+        List<Long> programIds = aclStore.getAccessiblePrograms( user.getUid(), Collections.emptyList() );
+
+        assertThat( programIds, hasSize( 0 ) );
+    }
+
+    @Test
+    public void getAccessibleProgramsReturnsPublicOnes()
+    {
+
+        // a publicly readable program
+        Program programA = createProgram( 'A' );
+        programA.getSharing().setOwner( owner );
+        programA.setPublicAccess( "--r-----" );
+        manager.save( programA, false );
+        // a private program
+        Program programB = createProgram( 'B' );
+        programB.getSharing().setOwner( owner );
+        programB.setPublicAccess( "--------" );
+        manager.save( programB, false );
+
+        List<Long> programIds = aclStore.getAccessiblePrograms( user.getUid(), Collections.emptyList() );
+
+        assertThat( programIds, hasSize( 1 ) );
+        assertThat( programIds, hasItem( programA.getId() ) );
```

Comment:
We have `Assertions.assertContainsOnly` in case you want to check a particular set of items but without order.